scripts/template-version-bump.js

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env node
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Checks git status for template changes and bumps version numbers
+// accordingly. Note that deletions do not cause version numbers to
+// be deleted, to ensure that we have sane metrics in the cases where
+// a deleted template is reinstated by some later commit or only one
+// format of a template is deleted.
+
+'use strict'
+
+const cp = require('child_process')
+const fs = require('fs')
+const path = require('path')
+
+const ROOT_DIR = path.join(__dirname, '..')
+const TEMPLATE_DIR = 'lib/senders/templates'
+const VERSIONS_FILE = '_versions.json'
+const IGNORE = new Set([ VERSIONS_FILE, '_pending.txt', 'index.js', 'README.md' ])
+const DEDUP = {}
+
+const templates = require(`../${TEMPLATE_DIR}`)
+const versions = require(`../${TEMPLATE_DIR}/${VERSIONS_FILE}`)
+
+cp.execSync('git status --porcelain', { cwd: ROOT_DIR, encoding: 'utf8' })
+  .split('\n')
+  .filter(line => line.match(`^[AM]. ${TEMPLATE_DIR}/\\w+`))
```

Comment:
Running this locally, I could not get it to detect changes and update the version file. Inspecting in debug mode, after the split above, there are extra spaces after the `M` git output. Using `^[AM ]. ${TEMPLATE_DIR}/\\w+` correctly wrote out the new versions.

This was with OSX bash and didn't have any special configs.

Reply:
Had you done `git add`? The script is supposed to ignore unstaged changes because they're not part of the current commit.

Reply:
That was it, worked as expected after adding 👍🏽

Reply:
Super duper! Thanks @vbudhram!